go-controller/pkg/ovn/loadbalancer/loadbalancer.go

```diff
@@ -36,41 +36,58 @@ import (
 // It is assumed that names are meaningful and somewhat stable, to minimize churn. This
 // function doesn't work with Load_Balancers without a name.
 func EnsureLBs(nbClient libovsdbclient.Client, externalIDs map[string]string, LBs []LB) error {
-	lbCache, err := GetLBCache(nbClient)
+	existing := libovsdbops.FindLoadBalancersByExternalIDs(nbClient, externalIDs)
```

Comment:
We need to check the performance impact of this, if we do pretend to solve the performance issue with this PR.
ref: https://bugzilla.redhat.com/show_bug.cgi?id=2035481

Reply:
As a reference, on this PR https://github.com/openshift/ovn-kubernetes/pull/900 we use the controller cache to know which LB don't already exist so we don't even go looking for them in the libovsdb cache and that seemed to fix the performance issue.

Reply:
See below for benchmarking results. You're right that this was a memory hog, until I fixed libovsdb.